core/src/lightclient/mod.rs

```diff
@@ -31,14 +60,25 @@ impl From<LightClientError> for jsonrpsee::core::Error {
     }
 }
 
-impl StarknetClient {
+impl StarknetClientImpl {
```

Comment:
I think it feels a bit bad to write implement an implementation

Reply:
Hahaha, we can change, but I dont see an issue

Reply:
alright let's keep it for now;D